fs/zbd_zenfs.h

```diff
@@ -127,6 +131,9 @@ class ZonedBlockDevice {
 
   std::mutex zone_resources_mtx_; /* Protects active/open io zones */
 
+  void SetMetrics(std::shared_ptr<ZenFSMetrics> metrics) { metrics_ = metrics; }
```

Comment:
I think we should drop this method and inject metric collection at construction time.